### ISSUE TITLE
Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by hhhhhumengshun** [baowenbo/DAIN#21]
UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details.
  "See the documentation of nn.Upsample for details.".format(mode))
Segmentation fault (core dumped)

Why？

**Comment by laomao0** [comment]
Refer to : https://github.com/princeton-vl/CornerNet-Lite/issues/7

In pytorch official tutorial, there are some clear explanation about this problem. https://pytorch.org/docs/stable/nn.html?highlight=upsample#torch.nn.Upsample
that is:
With align_corners = True, the linearly interpolating modes (linear, bilinear, bicubic, and trilinear) don’t proportionally align the output and input pixels, and thus the output values can depend on the input size. This was the default behavior for these modes up to version 0.3.1. Since then, the default behavior is align_corners = False. See below for concrete examples on how this affects the outputs.

**Comment by hhhhhumengshun** [comment]
How do you calculate ssim？ My calculations are always smaller than yours

**Comment by laomao0** [comment]
> How do you calculate ssim？ My calculations are always smaller than yours

python has different versions of ssim package, e.g., skimage , etc.
matlab also has function of ssim.
But matlab has higher ssim values than python.
typically，in one paper， we use one type of ssim function.
In this paper, I think the author use matlab function: ssim() to compute this metric of RGB image.

**Comment by duymanhdoan** [comment]
> UserWarning: Default upsampling behavior when mode=bilinear is changed to align_corners=False since 0.4.0. Please specify align_corners=True if the old behavior is desired. See the documentation of nn.Upsample for details. "See the documentation of nn.Upsample for details.".format(mode)) Segmentation fault (core dumped)
> 
> Why？

I solved problem.
the problem is opencv version base on c++, so that is Segmentation fault
just typed the commandline to change version of opencv:
$ pip3 install opencv-python==4.0.1.24
